### PR TITLE
using different urls "b2vapi.bmwgroup.com"

### PR DESCRIPTION
**Pull request by m1n3rva**
I checked out the code in #40 :+1: and the implementation at https://github.com/edent/BMW-i-Remote/ :+1: and did a refactoring to use this API in this library.

We could use the `b2vapi.*` urls to connect to the BMW servers and get the data. I guess this is what the smartphone apps are doing. This way we do not have to care about the different countries.

Maybe only differentiate between :us: , :canada: and the rest of the world. [The readme here](https://github.com/frankjoke/iobroker.bmw) lists also a url for China.

The interfaces on the server side are similar but not identical so I had to change several places. Now `status.py` and `remote_light_flash.py` are working for me (in Europe :eu: ) again. But most of the tests are broken, because the test data is now outdated.

Maybe this would also be the chance to integrate Minis!

What do you think?

### CODE REVIEW

File: bimmer_connected/account.py [hunk 1]
```diff
@@ -38,12 +38,13 @@ class ConnectedDriveAccount(object):  # pylint: disable=too-many-instance-attrib
     """
 
     # pylint: disable=too-many-arguments
-    def __init__(self, username: str, password: str, country: str, log_responses: str = None) -> None:
-        self._country = country
+    def __init__(self, username: str, password: str, region: str, log_responses: str = None) -> None:
```
Comment (ChristianKuehnel) [inline bimmer_connected/account.py]: shouldn't that be of type Region not str?

File: bimmer_connected/country_selector.py [hunk 2]
```diff
@@ -1,55 +1,27 @@
 """Get the right url for the different countries."""
+from enum import Enum
 import logging
-import requests
-from bimmer_connected.const import COUNTRY_SELECTION_URL
 
 _LOGGER = logging.getLogger(__name__)
 
 
+class Regions(Enum):
```
Comment (ChristianKuehnel) [inline bimmer_connected/country_selector.py]: please add China (as described in your comment)

File: bimmer_connected/country_selector.py [hunk 3]
```diff
@@ -1,55 +1,27 @@
...
+
+
+#: Mapping from regions to servers
+SERVER_URLS = {
+    Regions.NORTH_AMERICA: 'b2vapi.bmwgroup.us',
+    Regions.REST_OF_WORLD: 'b2vapi.bmwgroup.com',
+}
+
+
 class CountrySelector(object):  # pylint: disable=too-few-public-methods
```
Comment (ChristianKuehnel) [inline bimmer_connected/country_selector.py]: Maybe: Get rid of the class, as it does not contain much now anyway...

File: bimmer_connected/state.py [hunk 4]
```diff
@@ -87,17 +87,8 @@ def update_data(self) -> None:
         _LOGGER.debug('requesting new data from connected drive')
 
         response = self._account.send_request(
-            VEHICLE_STATE_URL.format(server=self._account.server_url, vin=self._vehicle.vin))
-
-        attributes = response.json()['attributesMap']
-        if attributes['head_unit'] not in ('NBTEvo', 'EntryEvo', 'NBT', 'EntryNav'):
```
Comment (ChristianKuehnel) [inline bimmer_connected/state.py]: can we add that check somewhere else? is there a `head_unit` attribute?

File: bimmer_connected/vehicle.py [hunk 5]
```diff
@@ -70,35 +69,3 @@ def __getattr__(self, item):
         """
         return self.attributes[item]
 
-
-class VehicleSpecs(object):  # pylint: disable=too-few-public-methods
```
Comment (ChristianKuehnel) [inline bimmer_connected/vehicle.py]: Why did you remove this? We should keep the functionality

File: remote_light_flash.py [hunk 6]
```diff
@@ -14,10 +14,9 @@ def main():
     parser.add_argument('username')
     parser.add_argument('password')
     parser.add_argument('vin')
-    parser.add_argument('country')
     args = parser.parse_args()
 
-    account = ConnectedDriveAccount(args.username, args.password, args.country)
+    account = ConnectedDriveAccount(args.username, args.password, Regions.REST_OF_WORLD)
```
Comment (ChristianKuehnel) [inline remote_light_flash.py]: please add command line argument for regions, so that users can test it

File: status.py [hunk 7]
```diff
@@ -14,10 +15,9 @@ def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('username')
     parser.add_argument('password')
-    parser.add_argument('country')
     args = parser.parse_args()
 
-    account = ConnectedDriveAccount(args.username, args.password, args.country)
+    account = ConnectedDriveAccount(args.username, args.password, Regions.REST_OF_WORLD)
```
Comment (ChristianKuehnel) [inline status.py]: please add command line argument for regions, so that users can test it

File: bimmer_connected/state.py (hunk 4 above)
Comment (m1n3rva) [reply bimmer_connected/state.py]: Sorry, there is not such attribute. There is however a "hmiVersion" attribute. Not sure how to map that to the head_unit types...

**Comment by m1n3rva** [comment]
I fixed the tests for the country selector :earth_asia: and disabled all the other tests until they are fixed.
And I did not find the `specs` in this API.

**Comment by ChristianKuehnel** [comment]
@gerard33 
What do you think? Should we switch to a different server API for better international support? I guess we still get all the data you see in the app. And this seems not to be identical with the web interface.

It would also mean to start collecting the data from the different vehicles again...

**Comment by gerard33** [comment]
@ChristianKuehnel 
I have just run the data with this new method and although all data seems to be available in there as well, it will require quite some rewriting off the module code (fields with different names, other way to check if vehicle tracking is enabled, GPS coordinates stored in a different way, no unit_of_lenght (all seems to be in [km](https://github.com/edent/BMW-i-Remote/#values-1), so no big issue), update time only shown in one format, etc.).

One of the fields in the vehicles.json is showing `"hmiVersion" : "ID5"` for my car, so maybe we can easily use that to determine if the car is [supported](https://community.home-assistant.io/t/bmw-connecteddrive-component/39569/179?u=gerard33) or not.

So if this is the best/only way to support all countries then I think we should switch.

And then we indeed need the collect the data again.

File: bimmer_connected/state.py (hunk 4 above)
Comment (gerard33) [reply bimmer_connected/state.py]: Maybe that `hmiVersion` is even better, see https://github.com/ChristianKuehnel/bimmer_connected/pull/43#issuecomment-372069868.

**Comment by ChristianKuehnel** [comment]
@gerard33 : yes I agree.

@m1n3rva 
I just created a new branch "b2vapi". Please rebase your PR on that and we'll do the work on the branch. Once the new API is working properly, we'll replace master with the b2vapi branch.

I would like to have a working version of the software on master in case anyone clones it...
